samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -44,7 +44,7 @@
 public class SamzaContainerController {
   private static final Logger log = LoggerFactory.getLogger(SamzaContainerController.class);
 
-  private final ExecutorService executorService;
```

Comment:
We may not manage multiple containers (at least, not as of today). However, based on the current ZkJobCoordinator implementation, we should be able to start and stop the container multiple times within the same lifecycle of the StreamProcessor. That said, this class can use some refactoring. I would like keep this outside the scope of this change. 
Ideally, I shouldn't have fixed this here either. So, I am going to revert back any unrelated changes in this class and fix them in the StreamProcessor/JobCoordinator refactoring.